homeassistant/components/zha/sensor.py

```diff
@@ -126,1504 +42,57 @@ async def async_setup_entry(
-        if cooling_demand is not None and cooling_demand > 0:
-            return HVACAction.COOLING
-
-        if self._cluster_handler.system_mode != self._cluster_handler.SystemMode.Off:
-            return HVACAction.IDLE
-        return HVACAction.OFF
-
-
-@MULTI_MATCH(
-    cluster_handler_names={CLUSTER_HANDLER_THERMOSTAT},
-    manufacturers="Sinope Technologies",
-    stop_on_match_group=CLUSTER_HANDLER_THERMOSTAT,
-)
-# pylint: disable-next=hass-invalid-inheritance # needs fixing
-class SinopeHVACAction(ThermostatHVACAction):
-    """Sinope Thermostat HVAC action sensor."""
-
-    @property
-    def _rm_rs_action(self) -> HVACAction:
-        """Return the current HVAC action based on running mode and running state."""
-
-        running_mode = self._cluster_handler.running_mode
-        if running_mode == self._cluster_handler.RunningMode.Heat:
-            return HVACAction.HEATING
-        if running_mode == self._cluster_handler.RunningMode.Cool:
-            return HVACAction.COOLING
-
-        running_state = self._cluster_handler.running_state
-        if running_state and running_state & (
-            self._cluster_handler.RunningState.Fan_State_On
-            | self._cluster_handler.RunningState.Fan_2nd_Stage_On
-            | self._cluster_handler.RunningState.Fan_3rd_Stage_On
-        ):
-            return HVACAction.FAN
-        if (
-            self._cluster_handler.system_mode != self._cluster_handler.SystemMode.Off
-            and running_mode == self._cluster_handler.SystemMode.Off
-        ):
-            return HVACAction.IDLE
-        return HVACAction.OFF
-
-
-@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_BASIC)
-# pylint: disable-next=hass-invalid-inheritance # needs fixing
-class RSSISensor(Sensor):
-    """RSSI sensor for a device."""
+            if entity_description.state_class is not None:
+                self._attr_state_class = SensorStateClass(
+                    entity_description.state_class.value
+                )
 
-    _attribute_name = "rssi"
-    _unique_id_suffix = "rssi"
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
-    _attr_device_class: SensorDeviceClass | None = SensorDeviceClass.SIGNAL_STRENGTH
-    _attr_native_unit_of_measurement: str | None = SIGNAL_STRENGTH_DECIBELS_MILLIWATT
-    _attr_entity_category = EntityCategory.DIAGNOSTIC
-    _attr_entity_registry_enabled_default = False
-    _attr_should_poll = True  # BaseZhaEntity defaults to False
-    _attr_translation_key: str = "rssi"
+            if entity_description.scale is not None:
+                self._attr_scale = entity_description.scale
 
-    @classmethod
-    def create_entity(
-        cls,
-        unique_id: str,
-        zha_device: ZHADevice,
-        cluster_handlers: list[ClusterHandler],
-        **kwargs: Any,
-    ) -> Self | None:
-        """Entity Factory.
+            if entity_description.native_unit_of_measurement is not None:
```

Comment:
This seems odd / different from Core.

A entity description does not win over the specific version. But considering it is set up later in ZHA, it means something different?

Also the naming is very confusing, as this is not about entity descriptions at all? At least, not our / Home Assistant's entity descriptions.